doc/tutorial/modes.txt

```diff
@@ -197,6 +197,33 @@ DebugMode      no         yes                VERY HIGH  Make many checks on what
 
 For more detail, see :ref:`Mode<libdoc_compile_mode>` in the library.
 
+.. _optimizers:
+
+Optimizers
+==========
+
+Theano allows compilations with a number of predefined optimizer modes.
+An optimizer mode consists of a particular set of optimizations, that speed
+up execution of Theano programs.
+
+The optimizers Theano provides are summarized below to indicate the trade-offs
+one might make between compilation time and execution time.
+
+=================  ============  ==============  ==================================================
+optimizer          Compile time  Execution time  Description
+=================  ============  ==============  ==================================================
+None               "++"          "++++++"        Applies none of Theano's opts
+o4 (fast_run)      "+++"         "+++++"         Applies all opts
+o3                 "++++"        "++++"          Applies all opts except inplace opts
+o2                 "+++++"       "+++"           Applies only few basic opts and fusion
+o1 (fast_compile)  "++++++"      "++"            Applies only few basic opts without fusion
+Stabilization      "+++++++"     "+"             Only applies softmax stability opt
+unsafe             "+"           "+++++++"       Applies all opts, and doesn't check input validity
```

Comment:
@nouiz In the same spirit, there could be in the future some unsafe optimizations that are unsafe for other reasons than not checking the input validity, no?